server/data/data-file/src/main/java/org/apache/james/mailrepository/file/MBoxMailRepository.java

```diff
@@ -690,4 +691,9 @@ public boolean lock(String key) {
     public boolean unlock(String key) {
         return false;
     }
+
+    @Override
+    public long size() throws MessagingException {
+        return Iterators.size(list());
```

Comment:
why don't you implement an internal method that returns an Iterable or a Collection or a List to avoid Iterators.size() that requires iterating on every elements ?

Reply:
No, it's directly in files, no memory data here... I don't want to invest more for this soon deprecated implementation.

Reply:
I don't agree, in list(), you have an ArrayList that implements size(). You don't need to iterate.